src/schema.ts

```diff
@@ -261,31 +271,56 @@ UserSchema.virtual("oldEnoughForWithdrawal").get(function (this: typeof UserSche
   return d - this.created_at.getTime() > yamlConfig.limits.oldEnoughForWithdrawal
 })
 
-UserSchema.methods.limitHit = async function ({
-  on_us,
-  amount,
-}: {
-  on_us: boolean
-  amount: number
-}) {
-  const timestampYesterday = Date.now() - MS_PER_DAY
+UserSchema.virtual("twoFactorEnabled").get(function (this: typeof UserSchema) {
+  return this.twoFactor.secret != null
+})
+
+const getTimestampYesterday = () => Date.now() - MS_PER_DAY
+
+// FIXME: Improve function name
+UserSchema.methods.remainingTwoFactorLimit = async function () {
```

Comment:
during the last code review (@krtk6160 you were not here but a recording should be available) @bodymindarts made a great point that those business logic function should not be part of the schema definition. we can move everything related to volume/velocity to UserWallet.ts, this would be a better place for this type of code.

Reply:
Yep, @bodymindarts and I had a discussion on the same about a different PR of mine in one of our pair programming sessions. We decided to not change schema.ts much because
1. The plan was to get my pending backend PRs merged quickly so I could switch to the release engineering stuff completely
2. Let the schema related refactor be part of the bigger upcoming refactor

I can refactor this in this PR itself, but it won't be a small refactor. I think it makes sense to do it in a separate PR. wdyt?

I can refactor this, or let it be like this depending on what makes sense.

Reply:
ok fair enough. let's not do this as part of this PR.